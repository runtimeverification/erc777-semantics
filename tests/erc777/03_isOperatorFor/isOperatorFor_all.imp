test.preamble();

// Testing isOperatorFor(), all possible scenarios.
//State:
//    - 2 token holders - 6 and 7
//    - 2 default operators: 8 and 9. Both authorized for 6, just 9 authorized for 7.
//    - 2 regular operators: 1 and 2. Both authorized for 6, just 2 authorized for 7.
//Test:
//    - isOperatorFor({1,2,8,9}, {6,7})

test.makeCaller(6);
test.authorizeOperator(1);
test.authorizeOperator(2);

test.makeCaller(7);
test.revokeOperator(8);
test.authorizeOperator(2);

test.isOperatorFor(1,6);
test.isOperatorFor(2,6);
test.isOperatorFor(8,6);
test.isOperatorFor(9,6);
test.isOperatorFor(1,7);
test.isOperatorFor(2,7);
test.isOperatorFor(8,7);
test.isOperatorFor(9,7);
