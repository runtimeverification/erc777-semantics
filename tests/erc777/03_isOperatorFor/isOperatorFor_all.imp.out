Changing the caller to 6
6 is authorizing 1 as an operator
6 is authorizing 2 as an operator
Changing the caller to 7
7 is revoking operator 8
7 is authorizing 2 as an operator
1 is operator for 6
2 is operator for 6
8 is operator for 6
9 is operator for 6
1 is not operator for 7
2 is operator for 7
8 is not operator for 7
9 is operator for 7
